electricitymap/contrib/lib/tests/test_event_lists.py

```diff
@@ -312,6 +312,92 @@ def test_merge_production_list_predicted(self):
         assert merged.events[0].storage.hydro == 2
         assert merged.events[0].sourceType == EventSourceType.forecasted
 
+    def test_merge_production_retains_corrected_negatives(self):
+        production_list_1 = ProductionBreakdownList(logging.Logger("test"))
+        production_list_1.append(
+            zoneKey=ZoneKey("AT"),
+            datetime=datetime(2023, 1, 1, tzinfo=timezone.utc),
+            production=ProductionMix(wind=-10, coal=10),
+            storage=StorageMix(hydro=1),
+            source="trust.me",
+        )
+        production_list_1.append(
+            zoneKey=ZoneKey("AT"),
+            datetime=datetime(2023, 1, 3, tzinfo=timezone.utc),
+            production=ProductionMix(wind=-12, coal=12),
+            storage=StorageMix(hydro=1, battery=1),
+            source="trust.me",
+        )
+        production_list_2 = ProductionBreakdownList(logging.Logger("test"))
+        production_list_2.append(
+            zoneKey=ZoneKey("AT"),
+            datetime=datetime(2023, 1, 1, tzinfo=timezone.utc),
+            production=ProductionMix(hydro=20, coal=20),
+            storage=StorageMix(hydro=1, battery=1),
+            source="trust.me",
+        )
+        production_list_2.append(
+            zoneKey=ZoneKey("AT"),
+            datetime=datetime(2023, 1, 3, tzinfo=timezone.utc),
+            production=ProductionMix(hydro=22, coal=22),
+            storage=StorageMix(hydro=1, battery=1),
+            source="trust.me",
+        )
+        merged = ProductionBreakdownList.merge_production_breakdowns(
+            [production_list_1, production_list_2],
+            logging.Logger("test"),
+        )
+        assert len(merged.events) == 2
+        assert merged.events[0].datetime == datetime(2023, 1, 1, tzinfo=timezone.utc)
+        assert merged.events[0].production.wind is None
+        assert merged.events[0].production.coal == 30
+        assert merged.events[0].storage.hydro == 2
+        assert merged.events[0].production._corrected_negative_values == {"wind"}
+
+        assert merged.events[1].datetime == datetime(2023, 1, 3, tzinfo=timezone.utc)
+        assert merged.events[1].production.wind is None
+        assert merged.events[1].production.coal == 34
+        assert merged.events[1].storage.hydro == 2
+        assert merged.events[1].production._corrected_negative_values == {"wind"}
```

Comment:
While not an issue per say all these declarations show type errors as production.* can be none and we don't explicitly do is not none first. Not sure if we need to "fix" this though but might be nice if we add typecheking to the CI later on.

Reply:
Added assertions to fix this.